File: src/main/java/ar/com/flow/bankaccount/withdrawal/LowerLimit.java
package ar.com.flow.bankaccount.withdrawal;

import ar.com.flow.bankaccount.Balance;
import ar.com.flow.bankaccount.BankAccount;
import ar.com.flow.money.Money;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class LowerLimit implements WithdrawalLimit {
    private final Balance limit;

    public boolean accepts(BankAccount account, Money amount) {
        var previewBalanceAfter = account.getBalance().minus(amount);

        return previewBalanceAfter.isGreaterThanOrEqual(limit);
    }
}

File: src/main/kotlin/ar/com/flow/bankaccount/withdrawal/LowerLimit.kt
package ar.com.flow.bankaccount.withdrawal

import ar.com.flow.bankaccount.BankAccount
import ar.com.flow.bankaccount.balance.Balance
import ar.com.flow.money.Money

class LowerLimit(private val limit: Balance) : WithdrawalLimit {
    override fun accepts(account: BankAccount, amount: Money): Boolean {
        val previewBalanceAfter = account.balance.minus(amount)
        return previewBalanceAfter.isGreaterThanOrEqual(limit)
    }
}